obsreport/obsreporttest/obsreporttest.go

```diff
@@ -48,41 +49,48 @@ var (
 type TestTelemetrySettings struct {
 	component.TelemetrySettings
 	SpanRecorder *tracetest.SpanRecorder
+	views        []*view.View
 }
 
+// ToExporterCreateSettings returns ExporterCreateSettings with configured TelemetrySettings
 func (tts *TestTelemetrySettings) ToExporterCreateSettings() component.ExporterCreateSettings {
 	exporterSettings := componenttest.NewNopExporterCreateSettings()
-	exporterSettings.TracerProvider = tts.TelemetrySettings.TracerProvider
+	exporterSettings.TelemetrySettings = tts.TelemetrySettings
 	return exporterSettings
 }
 
+// ToReceiverCreateSettings returns ReceiverCreateSettings with configured TelemetrySettings
 func (tts *TestTelemetrySettings) ToReceiverCreateSettings() component.ReceiverCreateSettings {
 	receiverSettings := componenttest.NewNopReceiverCreateSettings()
-	receiverSettings.TracerProvider = tts.TelemetrySettings.TracerProvider
+	receiverSettings.TelemetrySettings = tts.TelemetrySettings
 	return receiverSettings
 }
 
+// Shutdown unregisters any views and shuts down the SpanRecorder
+func (tts *TestTelemetrySettings) Shutdown() {
```

Comment:
Return error probably? Also may accept context (not sure, need ideas here)?

Reply:
updated to receive context and return an error. considering the main use is currently to defer shutdown, i'm not sure how often the error will be used, but since spanrecorder returns errors, its good to propagate it.